src/storage.ts

```diff
@@ -256,11 +260,18 @@ class AnnotationServerStorage {
                 Accept: "application/json",
                 "Content-Type": "application/json",
                 "X-CSRFToken": this.settings.csrf_token,
+                // ensure match with the previously fetched ETag
+                "If-Match": annotation.etag,
             },
             method: "POST",
         });
         if (res.ok) {
             return res.json();
+        } else if (res.status === 412) {
+            throw new Error(
+                `Error: Annotation was modified by another user while you were working.
+                Refresh the page to get the latest version, then make your changes.`,
```

Comment:
This one is interesting conceptually... feels a little weird to delete something that someone else just modified!

I guess the wording is vague enough here to indicate you might not want to delete it right away.

Is reloading the page the only way to refresh an annotation currently?  (probably fine, and seems simplest; just curious)

Reply:
Yes, that's the only way to refresh the annotation.

My thinking is exactly what you said—it would likely give the user a second thought about deleting since it clearly communicates that someone else worked on it. But it still gives instructions on how to delete if they want to.